skglm/tests/test_group.py

```diff
@@ -0,0 +1,126 @@
+import pytest
+
+import numpy as np
+from numpy.linalg import norm
+
+from skglm.penalties.block_separable import WeightedGroupL1
+from skglm.datafits.group import QuadraticGroup
+from skglm.solvers.group_bcd_solver import bcd_solver
+
+from skglm.utils import grp_converter, make_correlated_data
+from celer import GroupLasso, Lasso
+
+
+def _generate_random_grp(n_groups, n_features):
+    all_features = np.arange(n_features)
```

Comment:
It seems to me you can generate directly grp_ptr and grp_indices ? grp_indices is a permutation, grp_ptr is a np.random.choice(.. replace=False) that you sort and to which you add 0 and n_features

Because the first things we do after calling this function is to call grp_converter

Reply:
I agree with you but I think the current design enables us to condense 3 test configurations within one unit testing.

Reply:
what do you mean?